src/wallet/rpcwallet.cpp

```diff
@@ -3146,7 +3146,7 @@ UniValue setlabel(const JSONRPCRequest& request)
         if (pwallet->mapAddressBook.count(address.Get()))
         {
             std::string strOldLabel = pwallet->mapAddressBook[address.Get()].name;
-            if (address == GetAccountAddress(pwallet, strOldLabel)) {
+            if (strOldLabel != strLabel && address == GetAccountAddress(pwallet, strOldLabel)) {
```

Comment:
Note: This change makes calling setlabel on an address which already has the same label a no-op, see https://github.com/bitcoin/bitcoin/pull/7729#discussion_r92064042.